doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -149,74 +148,71 @@ always the intended equality for a given structure.
+   :g:`relation (A t1 ... tn)` over :g:`(A : αi -> ... αn -> Type)`.
 
-The :g:`@ident` gives a unique name to the morphism and it is used
-by the command to generate fresh names for automatically provided
-lemmas used internally.
+   The :token:`@ident` gives a unique name to the morphism and it is used
+   by the command to generate fresh names for automatically provided
+   lemmas used internally.
 
-Notice that the carrier and relation parameters may refer to the
-context of variables introduced at the beginning of the declaration,
-but the instances need not be made only of variables. Also notice that
-``A`` is *not* required to be a term having the same parameters as ``Aeq``,
-although that is often the case in practice (this departs from the
-previous implementation).
+   Notice that the carrier and relation parameters may refer to the
+   context of variables introduced at the beginning of the declaration,
+   but the instances need not be made only of variables. Also notice that
+   ``A`` is *not* required to be a term having the same parameters as ``Aeq``,
+   although that is often the case in practice (this departs from the
+   previous implementation).
 
+   To use this command, you need to first import the module ``Setoid`` using
+   the command ``Require Import Setoid``.
 
 .. cmd:: Add Relation
 
-In case the carrier and relations are not parametric, one can use this command
-instead, whose syntax is the same except there is no local context.
+   In case the carrier and relations are not parametric, one can use this command
+   instead, whose syntax is the same except there is no local context.
 
-The proofs of reflexivity, symmetry and transitivity can be omitted if
-the relation is not an equivalence relation. The proofs must be
-instances of the corresponding relation definitions: e.g. the proof of
-reflexivity must have a type convertible to
-:g:`reflexive (A t1 ... tn) (Aeq t′ 1 …t′ n )`.
-Each proof may refer to the introduced variables as well.
+   The proofs of reflexivity, symmetry and transitivity can be omitted if
+   the relation is not an equivalence relation. The proofs must be
+   instances of the corresponding relation definitions: e.g. the proof of
+   reflexivity must have a type convertible to
+   :g:`reflexive (A t1 ... tn) (Aeq t′ 1 …t′ n)`.
+   Each proof may refer to the introduced variables as well.
 
-**Example 6 (Parametric relation):**
+.. example:: Parametric relation
 
-For Leibniz equality, we may declare:
+   For Leibniz equality, we may declare:
 
-.. coqtop:: in
+   .. coqtop:: in
 
-  Add Parametric Relation (A : Type) : A (@eq A)
-    [reflexivity proved by @refl_equal A]
-  ...
+     Add Parametric Relation (A : Type) : A (@eq A)
+       [reflexivity proved by @refl_equal A]
+     ...
 
 Some tactics (:tacn:`reflexivity`, :tacn:`symmetry`, :tacn:`transitivity`) work only on
 relations that respect the expected properties. The remaining tactics
```

Comment:
Only spotted now: there is a ``` `replace` ``` on the line below that should become a ``` :tacn:`replace` ```

Reply:
Got 'em all.